src/index.d.ts

```diff
@@ -4363,6 +4363,29 @@ export default function createCachedSelector<S, P, R, T>(
   ParametricSelector<S, P, R>[]
 >;
 
+/*
+ * createStructuredCachedSelector
+ */
+type ValueOf<T> = T[keyof T];
```

Comment:
I would suggest naming this helper `$Values` and move to the top of the file, see https://github.com/piotrwitek/utility-types#valuest